build.sbt

```diff
@@ -99,19 +99,56 @@ headerSources in Compile ++= {
 lazy val root = (project in file("."))
 // Don't forget to add your sbt module here!
 // A missing module here can lead to failing Travis test results
-  .aggregate(`proxy-core`,
-             `proxy-cassandra`,
-             `proxy-postgres`,
-             `proxy-tests`,
-             `java-support`,
-             `scala-support`,
-             `java-shopping-cart`,
-             `akka-client`,
-             operator,
-             `tck`,
-             docs)
+  .aggregate(
+    `protocols`,
+    `proxy-core`,
+    `proxy-cassandra`,
+    `proxy-postgres`,
+    `proxy-tests`,
+    `java-support`,
+    `scala-support`,
+    `java-shopping-cart`,
+    `akka-client`,
+    operator,
+    `tck`,
+    docs
+  )
   .settings(common)
 
+lazy val protocols = (project in file("protocols"))
+  .settings(
+    name := "cloudstate-protos",
+    publish / skip := true,
+    packageBin in Compile := {
+      val base = baseDirectory.value
+      val commonProtos = base / (name.value + "-" + version.value + ".zip")
+      val tckProtos = base / (name.value + "-tck-" + version.value + ".zip")
+
+      def archiveStructure(topDirName: String, files: PathFinder): Seq[(File, String)] =
+        files pair Path.relativeTo(base) map {
+          case (f, s) => (f, s"$topDirName${File.separator}$s")
+        }
+
+      // Common Language Support Proto Dependencies
+      IO.zip(
+        archiveStructure("cloudstate-protocols",
+                         (base / "frontend" ** "*.proto" +++
+                         base / "protocol" ** "*.proto" +++
+                         base / "proxy" ** "*.proto")),
+        commonProtos
+      )
+
+      // Common TCK Language Support Proto Dependencies
+      IO.zip(archiveStructure("cloudstate-tck-protocols", base / "example" ** "*.proto"), tckProtos)
+
+      commonProtos
+    },
+    cleanFiles ++= Seq(
+        baseDirectory.value / (name.value + "-" + version.value + ".zip"),
```

Comment:
@viktorklang this would miss the version if the branch is updated, ok for you?

Reply:
@marcellanz I opted for being careful deleting files outside of the target directory. Perhaps we should not include the version number in the archive name at all?

Reply:
>Perhaps we should not include the version number in the archive name at all?

@viktorklang That makes sense I think. One would get the zip like so perhaps:
```
curl -O https://raw.githubusercontent.com/cloudstateio/cloudstate/v0.5.0/protocols/cloudstate-protos.zip
```
where its clear which version one gets. Same for every other tag or branch.

Reply:
:+1: Will amend.

Reply:
@marcellanz Done! See latest commit

Reply:
fetched and tested on my side, works! \o/

Reply:
@marcellanz Awesome! :-)